shared_error.go

```diff
@@ -75,6 +80,25 @@ func (s *SharedError) Store(err error) {
 	s.err = append(s.err, err)
 }
 
+// Return any error and reset the error if it was triggered.
+func (s *SharedError) Reset() error {
+	if s == nil {
+		return nil
+	}
+
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	err := s.string()
```

Comment:
Alternatively you could call `s.Error()` from the previous implementation. This method works fine though.